pkg/batch/cmd/server/server.go

```diff
@@ -594,37 +594,36 @@ func (s *server) startMultiPubSub(
 	processors *processor.Processors,
 	logger *zap.Logger,
 ) (*subscriber.MultiSubscriber, error) {
-	bytes, err := os.ReadFile(*s.subscriberConfig)
-	if err != nil {
-		logger.Error("subscriber: failed to read subscriber config",
-			zap.Error(err),
-		)
-		return nil, err
-	}
-	var configMap map[string]subscriber.Configuration
-	if err := json.Unmarshal(bytes, &configMap); err != nil {
-		logger.Error("subscriber: failed to unmarshal subscriber config",
-			zap.Error(err),
-		)
-		return nil, err
-	}
 	multiSubscriber := subscriber.NewMultiSubscriber(
 		subscriber.WithLogger(logger),
 	)
-	for name, config := range configMap {
-		p, err := processors.GetProcessorByName(name)
-		if err != nil {
-			logger.Error("subscriber: processor not found",
-				zap.String("name", name),
+	subscriberConfigBytes, err := os.ReadFile(*s.subscriberConfig)
+	if err != nil {
+		logger.Error("subscriber: failed to read subscriber config", zap.Error(err))
+	} else {
+		var configMap map[string]subscriber.Configuration
+		if err := json.Unmarshal(subscriberConfigBytes, &configMap); err != nil {
+			logger.Error("subscriber: failed to unmarshal subscriber config",
 				zap.Error(err),
 			)
 			return nil, err
 		}
-		multiSubscriber.AddSubscriber(subscriber.NewSubscriber(
-			name, config, p,
-			subscriber.WithLogger(logger),
-		))
+		for name, config := range configMap {
+			p, err := processors.GetProcessorByName(name)
+			if err != nil {
+				logger.Error("subscriber: processor not found",
+					zap.String("name", name),
+					zap.Error(err),
+				)
+				return nil, err
```

Comment:
We should also ignore it here.
It will fail if we add a new configuration, but it's not in the processors.
The idea is not to process persisters who don't have the configuration, or it's not specified in the processors.

Reply:
Typically, when we want to add a subscriber then we will add the subscriber config first. At the same time, the corresponding processor must be implemented and add in processor config.

Conclusion: all the subscribers `configMap`'s subscriber must have a processor, otherwise, we shouldn't start the batch service, because we or developer made a mistake that forgot to implement or add processor for the subscriber.

Reply:
If the container is deployed before the config map is updated, it will crash, and vice-versa.
We can't deploy both at the same time. So, this will be an issue, especially running canary releases and scaling out the pods simultaneously.

Reply:
In canary releases, it make sense.
Since we have this change in PR #952, I will update it there and close this PR.